### PR TITLE
P300Speller Game

### PR DESCRIPTION
**Pull request by HaowenWeiJohn**

### CODE REVIEW

File: Presets/LSLPresets/OpenBCILSL.json
Comment (ApocalyVec) [inline Presets/LSLPresets/OpenBCILSL.json:1]: Make a separate preset file for Cyton's P300 channel layout, as opposing overwriting the original channel layout which may create confusion to first-time users.

File: rena/examples/LSLExampleOutlet.py
Comment (ApocalyVec) [inline rena/examples/LSLExampleOutlet.py:1]: let's keep the LSLExampleOutlet consistently using 'Dummy-8Chan' as the channel name. Maybe create a separate script for obci_eeg1 to generate example stream if needed.

File: rena/scripting/Examples/P300SpellerDemo/P300SpellerScript.py
Comment (ApocalyVec) [inline rena/scripting/Examples/P300SpellerDemo/P300SpellerScript.py:1]: Add more docstrings to explain what this script is doing, as this is the script the user will see the most. 

For example, add docstring for each state in the loop function and each declaration of variable.

This comment in the cleanup call:
        # event marker followed by horizontal and vertical index
        # ................|..................|(processed marker).................|...................|............
        # ....|....|....|....|....|....|....|....|....|....|....|....|....|....|....|....|....|....|
is very good for showing how the event marker works. Add some thing similar to this in the class docstirng. It should show what event marker are received during a session.

File: rena/scripting/Examples/P300SpellerDemo/P300Speller_params.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+###########################################
+IDLE_STATE = 0
+RECORDING_STATE = 1
+TRAINING_STATE = 99
+TESTING_STATE = 100
+############################################
+
+EEG_SAMPLING_RATE = 250.0
+
+Time_Window = 1.1  # second
```
Comment (ApocalyVec) [inline rena/scripting/Examples/P300SpellerDemo/P300Speller_params.py:37]: I suggest put 'second' in the variable name so it becomes "time_window_seconds"

File: rena/scripting/Examples/P300SpellerDemo/P300Speller_params.py
Comment (ApocalyVec) [inline rena/scripting/Examples/P300SpellerDemo/P300Speller_params.py:1]: add a header docstring to this file, explain in summary what parameters are listed here and what they do.

File: rena/scripting/Examples/P300SpellerDemo/P300Speller_utils.py
Comment (ApocalyVec) [inline rena/scripting/Examples/P300SpellerDemo/P300Speller_utils.py:1]: add docstring to each of the functions in this file. User may want to reuse some of these well-written functions in their pipeline.

File: rena/scripting/Examples/P300SpellerDemo/offline_analysis.py
Comment (ApocalyVec) [inline rena/scripting/Examples/P300SpellerDemo/offline_analysis.py:1]: add comments to this script explain how it should run if we were to include it in the release. You can duplicate stuff from the docs.

**Review by ApocalyVec — CHANGES_REQUESTED** [review]
Very good and well-written stuff, most of the changes requested center around adding docstrings. Please make do so and we can include it in the next release

File: rena/examples/LSLExampleOutlet.py
Comment (HaowenWeiJohn) [reply rena/examples/LSLExampleOutlet.py:1]: The LSLExampleOutlet stream name has been changed back to Dummy-8Chan

File: Presets/LSLPresets/OpenBCILSL.json
Comment (HaowenWeiJohn) [reply Presets/LSLPresets/OpenBCILSL.json:1]: Fixed

File: rena/examples/LSLExampleOutlet.py
Comment (HaowenWeiJohn) [reply rena/examples/LSLExampleOutlet.py:1]: Fixed

File: rena/scripting/Examples/P300SpellerDemo/P300Speller_params.py (hunk 1 above)
Comment (HaowenWeiJohn) [reply rena/scripting/Examples/P300SpellerDemo/P300Speller_params.py:37]: Fixed

**Review by ApocalyVec — APPROVED** [review]

**Pull request merged**